src/schema.jl

```diff
@@ -53,6 +53,19 @@ Base.merge!(a::Schema, b::Schema) = (merge!(a.schema, b.schema); a)
 Base.keys(schema::Schema) = keys(schema.schema)
 Base.haskey(schema::Schema, key) = haskey(schema.schema, key)
 
+function Base.:(==)(first::Schema, second::Schema)
+    first === second && return true
+    first.schema === second.schema && return true
+    length(first.schema) != length(second.schema) && return false
+    for key in keys(first)
+        !haskey(second, key) && return false
+        second[key] != first[key] && return false
+    end
+    true
+end
+
+Base.hash(schema::Schema, h::UInt) = hash(schema.schema, h)
```

Comment:
Sometimes (e.g. for arrays and tuples) we add an arbitrary constant to `h` (type-specific) to ensure that `hash(schema) != hash(schema.schema)`. Not sure whether it's worth it here though.

Reply:
And that can "corrupt" the type of containers because of the *ahem* zealous use of type parameters in this codebase

Reply:
You don't necessarily need to hash the type: you can just define a constant and use it for all types which compare `==`.